### DISCUSSION TITLE
Extending Docker image fails when adding build-essential

### DISCUSSION THREAD

**Discussion by davidgin** [apache/airflow#21896] [General]
### Apache Airflow version

2.2.4 (latest released)

### What happened

When extending the docker image with  build-essential
I added according to the [documentation ](https://airflow.apache.org/docs/docker-stack/build.html#customizing-the-image) to my DockerFile .
```
I get the following error : Setting up libc6:amd64 (2.33-7) ...
/usr/bin/perl: error while loading shared libraries: libcrypt.so.1: cannot open shared object file: No such file or directory
dpkg: error processing package libc6:amd64 (--configure):
 installed libc6:amd64 package post-installation script subprocess returned error exit status 127
Errors were encountered while processing:
 libc6:amd64
E: Sub-process /usr/bin/dpkg returned an error code (1)
The command '/bin/bash -o pipefail -o errexit -o nounset -o nolog -c apt-get update  && apt-get install -y --no-install-recommends  build-essential  && apt-get autoremove -yqq --purge  && apt-get clean  && rm -rf /var/lib/apt/lists/*' returned a non-zero code: 100

```
My DockerFile is 
`:FROM dockerhub.artifactory..****/apache/airflow:2.2.4
USER root
RUN sh -c "echo >  /etc/apt/sources.list"
RUN sh -c "echo 'deb-src https://artifactory.******/artifactory/debian/ buster main' >> /etc/apt/sources.list"
RUN sh -c "echo 'deb https://artifactory.*****/artifactory/mysql-debian/ buster mysql-8.0' >> /etc/apt/sources.list"
RUN sh -c "echo 'deb https://artifactory.*******/artifactory/debian/ sid main' >> /etc/apt/sources.list"
RUN sh -c "echo 'deb https://artifactory.****/artifactory/debian-packages/ buster main' >> /etc/apt/sources.list"
RUN sh -c "echo 'deb https://artifactory*****/artifactory/debian-packages/ buster libc6' >> /etc/apt/sources.list"
**RUN apt-get update \
 && apt-get install -y --no-install-recommends \
 build-essential \**
 && apt-get autoremove -yqq --purge \
 && apt-get clean \
 && rm -rf /var/lib/apt/lists/*
USER airflow
RUN pip install --no-cache-dir --user apache-airflow-providers-apache-hive==2.1.0 -i https://artifactory.*****/artifactory/api/pypi/pypi-remote/simple
`
This seems to related to a [bug](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=974552) in debian distribution .

### What you expected to happen

I expect build-essential package to be unpacked and installed successfully.

The log I get `
Setting up libc6:amd64 (2.33-7) ...
/usr/bin/perl: error while loading shared libraries: libcrypt.so.1: cannot open shared object file: No such file or directory
dpkg: error processing package libc6:amd64 (--configure):
 installed libc6:amd64 package post-installation script subprocess returned error exit status 127
Errors were encountered while processing:
 libc6:amd64
E: Sub-process /usr/bin/dpkg returned an error code (1)
The command '/bin/bash -o pipefail -o errexit -o nounset -o nolog -c apt-get update  && apt-get install -y --no-install-recommends  build-essential  && apt-get autoremove -yqq --purge  && apt-get clean  && rm -rf /var/lib/apt/lists/*' `
 

### How to reproduce

(I use an internal artifactory mirror but the bug persists also when running locally  )
Build the docker image according to official documentation.

docs/docker-stack/docker-examples/extending/add-build-essential-extend/Dockerfile

FROM apache/airflow:2.2.4
USER root
RUN apt-get update \
  && apt-get install -y --no-install-recommends \
         **build-essential** libopenmpi-dev \
  && apt-get autoremove -yqq --purge \
  && apt-get clean \
  && rm -rf /var/lib/apt/lists/*
USER airflow
RUN pip install --no-cache-dir mpi4py
 

### Operating System

The host is CentOS Linux release 7.9.2009 (Core). 

### Versions of Apache Airflow Providers

apache-airflow-providers-apache-hive==2.1.0

### Deployment

Other Docker-based deployment

### Anything else

Happens Every time

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
Build essentials install properly.

 I could not reproduce it with this Dockerfile you mentioned, it all works as expected (and actually it is run with every PR of ours on our CI so it SHOULD work).

```
FROM apache/airflow:2.2.4
USER root
RUN apt-get update && apt-get install -y --no-install-recommends build-essential libopenmpi-dev && apt-get autoremove -yqq --purge && apt-get clean && rm -rf /var/lib/apt/lists/*
USER airflow
RUN pip install --no-cache-dir mpi4py
```
Result:

```
Successfully built mpi4py
Installing collected packages: mpi4py
Successfully installed mpi4py-3.1.3
WARNING: You are using pip version 21.3.1; however, version 22.0.3 is available.
You should consider upgrading via the '/usr/local/bin/python -m pip install --upgrade pip' command.
Removing intermediate container 57c326cdf76d
 ---> 2f7846c80a21
Successfully built 2f7846c80a21
```

You have other things in your dockerfile - especially puling libc. This is a problem with your Dockerfile, not Airlfow.

**Reply by potiuk** [reply]
Converted it in to discussion as you might want to provide results of your investigations on what was wrong.

**Reply by potiuk** [reply]
According to this https://www.mail-archive.com/debian-bugs-dist@lists.debian.org/msg1818037.html - you will get into this problem when you have conflicting version of libc and libcrypt and since you are actually trying to install another version of libc manually. This is definitely a problem that your installation triggered.
